### PR TITLE
remove firebase support from dartpad

### PR DESCRIPTION
**Pull request by devoncarew**
- remove firebase support from dartpad

Starting as a draft as this will require discussion, comms, and some planning.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by parlough** [comment]
Thanks for opening this!

Do consider removing the Firebase sample from the dropdown as a first step that could land separately/earlier. It's what clearly indicates Firebase is supported currently.

File: pkgs/dart_services/lib/src/analysis.dart [hunk 1]
```diff
@@ -327,7 +325,16 @@ class AnalysisServerWrapper {
         }
       } else if (import.packageImport) {
         final packageName = import.packageName;
-        if (!isSupportedPackage(packageName)) {
+
+        if (isFirebasePackage(packageName)) {
+          // TODO: Update the 'more info' url.
```
Comment (devoncarew) [inline pkgs/dart_services/lib/src/analysis.dart:330]: @johnpryan - do you have a link we should use here? Either a github issue or a wiki page, likely w/ context about the features changes for the dartpad refresh.

Comment (johnpryan) [reply pkgs/dart_services/lib/src/analysis.dart:330]: It looks like @parlough added details here: https://github.com/dart-lang/dart-pad/wiki/Package-and-plugin-support#deprecated-firebase-packages

**Review by johnpryan — APPROVED** [review]

File: pkgs/dart_services/lib/src/analysis.dart (hunk 1 above)
Comment (parlough) [reply pkgs/dart_services/lib/src/analysis.dart]: Thanks! Feel free to adjust or add to the messaging there.

**Comment by parlough** [comment]
@devoncarew I noticed this also removed support for the Flame packages. Is that a goal or were they causing issues?

**Comment by devoncarew** [comment]
> @devoncarew I just noticed this also removed support for the Flame packages. Is that a goal or were they causing any particular issue?

Negative - that was not a goal. I suspect I confused them w/ firebase related packages.

**Comment by devoncarew** [comment]
I'll look into re-adding these packages.

**Comment by parlough** [comment]
> I'll look into re-adding these packages.

Thanks! Let me know if you have any issues.

Fire and flame do seem similar 🔥

**Pull request merged**